src/loading/binary-loader.ts

```diff
@@ -93,8 +93,15 @@ export class BinaryLoader {
     return url;
   }
 
-  private parse(node: PointCloudOctreeGeometryNode, buffer: ArrayBuffer): void {
+  private parse(node: PointCloudOctreeGeometryNode, buffer: ArrayBuffer): Promise<void> {
+    return new Promise((resolve) => {
+      this._parse(node, buffer, resolve);
+    });
+  }
+
+  private _parse(node: PointCloudOctreeGeometryNode, buffer: ArrayBuffer, resolve: () => void) {
```

Comment:
Why not keep this as `parse` and then just do this when loading:

```ts
return Promise.resolve(this.getUrl(this.getNodeUrl(node)))
  .then(url => this.xhrRequest(url, { mode: 'cors' }))
  .then(res => res.arrayBuffer())
  .then(buffer => {
    return new Promise(resolve => this.parse(node, buffer, resolve));
  });
```

Reply:
Nice :)